declarative-examples/buildPluginParentPOM.groovy

```diff
@@ -0,0 +1,29 @@
+pipeline {
+    tools {
+        maven "apache-maven-3.1.0"
+        jdk "default"
+    }
+
+    agent {
+        label "master"
+    }
+
+    stages {
+        stage("build") {
+            steps {
+                dir("tmp") {
+                    git changelog: false, poll: false, url: 'git://github.com/jenkinsci/plugin-pom.git', branch: 'master'
+                    sh 'echo "M2_HOME: ${M2_HOME}"'
```

Comment:
There could be a `mvn` step, but that'd have to be written elsewhere - that's not something Declarative can provide.